database/changesets/20210709_TerraDataRepoSink.xml

```diff
@@ -0,0 +1,71 @@
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+                        http://www.liquibase.org/xml/ns/dbchangelog
+                        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+    <changeSet author="hornet@broadinstitute.org"
+               dbms="postgresql"
+               id="20210709"
+               logicalFilePath="20210709_TerraDataRepoSink.xml"
+               runInTransaction="false">
+        <comment>
+            A descriptor for the type of TDR job wfl started while ingesting
+            outputs
+        </comment>
+        <sql>
+            CREATE TYPE TDRJobType AS ENUM (
+                'FileLoad',
+                'MetadataIngest'
+            )
+        </sql>
+        <comment>
+            The status of the TDR job wfl started, as reported by TDR. See
+            https://data.terra.bio/swagger-ui.html#/repository/retrieveJob
+            for details.
+        </comment>
+        <sql>
+            CREATE TYPE TDRJobStatus AS ENUM (
+                'running',
+                'succeeded',
+                'failed'
+            )
+        </sql>
+        <comment>
+            The TDR Job Queue used by the `TerraDataRepo` Sink
+        </comment>
+        <sql>
+            CREATE TYPE TerraDataRepoSinkDetails AS (
+                id        BIGINT,       -- ALTER ADD GENERATED ALWAYS AS IDENTITY
+                type      TDRJobType,   -- For dispatching to right handler
+                job       TEXT,         -- UUID of the job created in TDR
+                status    TDRJobStatus, -- As reported by TDR
+                updated   TIMESTAMPTZ,  -- When this record was last updated
+                payload   TEXT,         -- generic JSON payload for the handler
+                consumed  TIMESTAMPTZ   -- when this job was "evicted" from the queue
+            )
+        </sql>
+        <comment>
+            the `TerraDataRepo` sink table schema
+        </comment>
+        <sql>
+            CREATE TABLE TerraDataRepoSink (
+                id            BIGINT GENERATED BY DEFAULT AS IDENTITY NOT NULL,  -- primary key
+                dataset       TEXT NOT NULL,                                     -- the serialized EDN of the TDR dataset schema
+                dataset_table TEXT NOT NULL,                                     -- the name of the dataset table
+                from_outputs  TEXT NOT NULL,                                     -- mapping of how to coerce the output of executor to whatever the sink understands in serialized EDN
+                details       TEXT NOT NULL,                                     -- the name of the sink details table
+                CONSTRAINT TerraDataRepoSink_PKEY PRIMARY KEY (id)
```

Comment:
Nit: other existing primary key constraints are defined in capslock, i.e. `TERRADATAREPOSINK_PKEY`.  Not sure if that was a conscious choice though.

Reply:
Thanks - it's good to be consistent!